osu.Game/Scoring/Legacy/LegacyScoreEncoder.cs

```diff
@@ -93,7 +93,7 @@ public void Encode(Stream stream, bool leaveOpen = false)
                 sw.Write((ushort)(score.ScoreInfo.GetCountMiss() ?? 0));
                 sw.Write((int)(score.ScoreInfo.TotalScore));
                 sw.Write((ushort)score.ScoreInfo.MaxCombo);
-                sw.Write(score.ScoreInfo.Combo == score.ScoreInfo.MaxCombo);
+                sw.Write(score.ScoreInfo.Combo == score.ScoreInfo.GetMaximumAchievableCombo());
```

Comment:
I'm.... not even sure what wass happening here? The old implementation looks completely wrong and so does this one for that matter.

Why was this comparing two values from `ScoreInfo` to begin with? Why is `Combo` in the comparison at all, if it appears to be the _current running combo_ for _spectator purposes_? Is this even guaranteed to be populated correctly in the database? In mine I have several scores matching `Combo != MaxCombo`.

Reply:
You're right that this should've been `MaxCombo`. Though I would hope `Combo == MaxCombo` in an `IsPerfect == true` scenario, otherwise that sounds like an issue.